protocols/primary-backup/src/main/java/io/atomix/protocols/backup/service/impl/PrimaryBackupServiceContext.java

```diff
@@ -299,7 +299,6 @@ public void resetIndex(long index, long timestamp) {
     currentOperation = OperationType.COMMAND;
     operationIndex = index;
     currentIndex = index;
-    commitIndex = index;
```

Comment:
I think this just need to be updated, it just can't be decreased.

Reply:
ohh you mean like this? commitIndex = Math.max(commitIndex, index);

Reply:
yeah... or `setCommitIndex` has the `max` call in it already